app/src/main/java/grinnell/appdev/edu/lyles/fragments/FavoritesFragment.java

```diff
@@ -1,22 +1,139 @@
+
+    private FavoritesManager mFavoritesManager;
+
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
-        return inflater.inflate(R.layout.favorites_layout, container, false);
+        View view = inflater.inflate(favorites_layout, container, false);
+
+        setUpConstantArrays();
+        mAllMenuItems = getMenuItems(mAllURLs, mAllArrayTitles);
+
+        mFavoritesManager = new FavoritesManager(getContext(), mAllMenuItems);
+        mItemAdapter = new ItemAdapter(this.getContext(), mFavoritesManager.getAllFavorites(), true);
+
+        mRecyclerView = (RecyclerView) view.findViewById(R.id.rv_items_favorites);
+        mRecyclerView.setAdapter(mItemAdapter);
+        mRecyclerView.setLayoutManager(new LinearLayoutManager(this.getContext()));
+
+        return view;
+    }
+
+    @Override
+    public void onDestroyView() {
+        mAllURLs = null;
+        mAllArrayTitles = null;
+
+        mAllMenuItems = null;
+        mItemAdapter = null;
+        mRecyclerView = null;
+
+        mFavoritesManager = null;
+
+        super.onDestroyView();
+    }
+
+    @Override
+    public void onDestroy() {
+        mAllURLs = null;
+        mAllArrayTitles = null;
+
+        mAllMenuItems = null;
+        mItemAdapter = null;
+        mRecyclerView = null;
+
+        mFavoritesManager = null;
+
+        super.onDestroy();
+    }
+
+    /**
+     * Adds all constants corresponding urls and array keys into arrays
+     */
+
+    private void setUpConstantArrays() {
+        // URLs of JSON arrays to use
+        mAllURLs = new ArrayList<String>();
+        mAllURLs.add(Constants.HOT_FOOD_URL);
+        mAllURLs.add(Constants.SNACKS_URL);
+        mAllURLs.add(Constants.DRINKS_URL);
+        mAllURLs.add(Constants.BEER_URL);
+
+        // Titles of JSON arrays in corresponding urls
+        mAllArrayTitles = new ArrayList<String>();
+        mAllArrayTitles.add(Constants.HOT_FOOD_ARRAY_KEY);
+        mAllArrayTitles.add(Constants.SNACKS_ARRAY_KEY);
+        mAllArrayTitles.add(Constants.DRINKS_ARRAY_KEY);
+        mAllArrayTitles.add(Constants.BEER_ARRAY_KEY);
     }
 
+    /**
+     * Using a list of urls and json keys, retrieves a jsonArray asynchronously and returns a list of all MenuItems
+     *
+     * @param urls an ArrayList of urls as Strings
+     * @param keys an ArrayList of keys for parsing the Json Array as strings
+     * @return     an ArrayList of type MenuItem containing all items found from all urls
+     */
+
+    private ArrayList<MenuItem> getMenuItems(ArrayList<String> urls, ArrayList<String> keys) {
+
+        ArrayList<MenuItem> returnList = new ArrayList<>();
+
+        for(int i = 0; i < urls.size(); i++) {
+            AsyncRetrieval asyncRetrieval = new AsyncRetrieval(urls.get(i));
+
+            String jsonBody = "";
+            JSONArray jsonArray = new JSONArray();
+
+            try {
+                jsonBody = asyncRetrieval.execute().get();
```

Comment:
This will start the `AsyncTask` in the background and then waits on the main thread, due to `get`, which defeats the purpose of running things in the background. Instead, when you create the `AsyncRetrieval` do something like this:
```java
AsyncRetrieval asyncRetrieval = new AsyncRetrieval(urls.get(i)) {
    @Override
    protected void onPostExecute(String result) {
        // do something with the data
    }
}
```